File: src/main/kotlin/ru/exrates/entities/exchanges/secondary/Limit.java
package ru.exrates.entities.exchanges.secondary;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import ru.exrates.repos.DurationConverter;

import javax.persistence.*;
import java.time.Duration;

@Entity
@Table(name = "Limits")
@NoArgsConstructor
public class Limit {
    @Id
    @GeneratedValue
    private Integer id;
    @Getter @Setter private String name;
            @Column(name = "_interval") @Convert(converter = DurationConverter.class)
            @Getter private Duration interval;
            @Enumerated(value = EnumType.STRING)
            @Column(length = 20)
            @Getter private LimitType type;
    @Getter @Setter private int limitValue;


    public Limit(String name, LimitType type, Duration interval, int limit) {
        this.name = name;
        this.type = type;
        this.limitValue = limit;
        this.interval = interval;
    }

}

File: src/main/kotlin/ru/exrates/entities/exchanges/secondary/Limit.kt
package ru.exrates.entities.exchanges.secondary

import ru.exrates.entities.LimitType
import ru.exrates.repos.DurationConverter
import java.time.Duration
import javax.persistence.*

@Entity
@Table(name = "Limits")
class Limit(){
    lateinit var name: String
    @Column(length = 20) @Enumerated(value = EnumType.STRING)
    lateinit var type: LimitType
    @Convert(converter = DurationConverter::class) @Column(name = "_interval")
    lateinit var interval: Duration
    var limitValue: Int = 0
    @Id @GeneratedValue
    private val id: Int = 0

    constructor(name: String, type: LimitType, interval: Duration, limitValue: Int) : this(){
        this.name = name
        this.interval = interval
        this.limitValue = limitValue
        this.type = type
    }
}
    
